drivers/sensor/ens210/ens210.c

```diff
@@ -46,6 +46,39 @@ static int ens210_sample_fetch(struct device *dev, enum sensor_channel chan)
 	u32_t temp_valid, humidity_valid;
 #endif /* CONFIG_ENS210_CRC_CHECK */
 
+#if defined(CONFIG_ENS210_TEMPERATURE_SINGLE) \
+	|| defined(CONFIG_ENS210_HUMIDITY_SINGLE)
+
+	const struct ens210_sens_start sense_start = {
+		.t_start = ENS210_T_START,
+		.h_start = ENS210_H_START
+	};
+
+	/* Start measuring */
+	ret = i2c_reg_write_byte(drv_data->i2c,
+			DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+			ENS210_REG_SENS_START, *(u8_t *)&sense_start);
+	if (ret < 0) {
+		LOG_ERR("Failed to set SENS_START to 0x%x",
+				*(u8_t *)&sense_start);
+		return -EIO;
+	}
+
+	u8_t buf;
+
+	do {
+		k_sleep(K_MSEC(2));
+		ret =  i2c_reg_read_byte(drv_data->i2c,
+				DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+				ENS210_REG_SENS_START, &buf);
```

Comment:
You should read ENS210_REG_SENS_STAT her right?
From the datasheet:
```
Address 0x24 SENS_STAT
indicates the measuring status of the relative humidity sensor
0: Relative humidity sensor is idle (not measuring)
1: Relative humidity sensor is active measuring
```

Reply:
Well, if we want to fetch the data from the `ENS210_REG_SENS_STAT` register it makes everything a little bit more complicated. If you check the figure 18 on page 16 in the [datasheet](https://ams.com/documents/20143/36005/ENS210_DS000470_3-00.pdf) you see that the `ENS210_REG_SENS_STAT` register goes to hight after the sensor booted. This means that before checking the register I need to check the boot state. On the other side, if I use only the `ENS210_REG_SENS_START` register I can skip this part because it also goes to low as soon as the measurement finishes.